components/dashboard/components/CourseConnections/CourseConnectionsView.jsx

```diff
@@ -210,9 +210,17 @@ class CourseConnectionsView extends React.Component {
      * Draw the graph w/ the latest interaction data
      */
     drawGraph() {
-        const {userInteractions, userLearningGroups} = this.props;
+        const {userLearningGroups} = this.props;
         const learningGroupTypes = this.sortedLearningGroupTypes;
 
+        /** DevNote: This is a temporary solution that filters direct messages sent by 'riffbot'
+         *          out of the user interactions. This is to prevent them from appearing on the
+         *          course connections graph. In the future, the user interactions query in
+         *          mattermost-server will be refactored, and will account for this, and we can
+         *          remove this filter at that point.
+         */
+        const userInteractions = [...this.props.userInteractions].filter((interaction) => interaction.Username !== 'riffbot');
```

Comment:
what's the reasoning behind creating a copy of the array before the filter?

Reply:
Oh I guess this technically doesn't matter, because filter creates a new array anyway..it's non-mutating?

Reply:
Removed.